test/purchases_flutter_test.dart

```diff
@@ -175,4 +175,60 @@ void main() {
       fail("there was an exception " + e.toString());
     }
   });
+
+  test('errors are mapped correctly', () {
+    expect(PurchasesErrorCode.unknownError.index, 0);
+    expect(PurchasesErrorCode.purchaseCancelledError.index, 1);
+    expect(PurchasesErrorCode.storeProblemError.index, 2);
+    expect(PurchasesErrorCode.purchaseNotAllowedError.index, 3);
+    expect(PurchasesErrorCode.purchaseInvalidError.index, 4);
+    expect(PurchasesErrorCode.productNotAvailableForPurchaseError.index, 5);
+    expect(PurchasesErrorCode.productAlreadyPurchasedError.index, 6);
+    expect(PurchasesErrorCode.receiptAlreadyInUseError.index, 7);
+    expect(PurchasesErrorCode.invalidReceiptError.index, 8);
+    expect(PurchasesErrorCode.missingReceiptFileError.index, 9);
+    expect(PurchasesErrorCode.networkError.index, 10);
+    expect(PurchasesErrorCode.invalidCredentialsError.index, 11);
+    expect(PurchasesErrorCode.unexpectedBackendResponseError.index, 12);
+    expect(PurchasesErrorCode.receiptInUseByOtherSubscriberError.index, 13);
+    expect(PurchasesErrorCode.invalidAppUserIdError.index, 14);
+    expect(PurchasesErrorCode.operationAlreadyInProgressError.index, 15);
+    expect(PurchasesErrorCode.unknownBackendError.index, 16);
+    expect(PurchasesErrorCode.invalidAppleSubscriptionKeyError.index, 17);
+    expect(PurchasesErrorCode.ineligibleError.index, 18);
+    expect(PurchasesErrorCode.insufficientPermissionsError.index, 19);
+    expect(PurchasesErrorCode.paymentPendingError.index, 20);
+    expect(PurchasesErrorCode.invalidSubscriberAttributesError.index, 21);
+    expect(PurchasesErrorCode.logOutWithAnonymousUserError.index, 22);
+    expect(PurchasesErrorCode.configurationError.index, 23);
+    expect(PurchasesErrorCode.unsupportedError.index, 24);
+  });
+
+  test('PurchasesErrorHelper maps errors correctly', () {
+    expect(PurchasesErrorHelper.getErrorCode(PlatformException(code: '0')), PurchasesErrorCode.unknownError);
+    expect(PurchasesErrorHelper.getErrorCode(PlatformException(code: '1')), PurchasesErrorCode.purchaseCancelledError);
+    expect(PurchasesErrorHelper.getErrorCode(PlatformException(code: '2')), PurchasesErrorCode.storeProblemError);
+    expect(PurchasesErrorHelper.getErrorCode(PlatformException(code: '3')), PurchasesErrorCode.purchaseNotAllowedError);
+    expect(PurchasesErrorHelper.getErrorCode(PlatformException(code: '4')), PurchasesErrorCode.purchaseInvalidError);
```

Comment:
was wondering if there's any way for us to check that the length of PurchasesErrorCode is equal to the length in the native SDKs, but i don't think we expose those anywhere? just throwing it out there in case you have ideas. could help us catch missing error codes in the future.

Reply:
I'm really not sure how we could do that, since all calls to the native stuff go through `channel`, and the channel for tests isn't really tied to native code.

I think the closest we might be able to get is to create a test suite for android and one for iOS, then have a test that does an exhaustive switch for all existing cases in the enum, no default. 
if a new case gets added to the enum, the compiler should complain. 

I think that's outside the scope of this PR, but I'll throw something on the backlog for it